model.py

```diff
@@ -1,103 +1,1394 @@
+
+        if save:
+            # Check that save_dir exists
+            if not os.path.isdir(save_dir):
+                os.mkdir(save_dir)
+
+            # If we didnt get told to use a certain fname, use this node's name
+            if fname is None:
+                fname = self.lc.name.replace('.calib', '.pdf')
+
+            # Make the filename
+            fname = '/'.join([save_dir, fname])
+
+            # If the user specified a path like './figs/', then the above could
+            # return './figs//Eclipse_N.pdf'; I want to be robust against that.
+            while '//' in fname:
+                fname = fname.replace('//', '/')
+
+            plt.savefig(fname)
+
+        return fig, axs
+
+    @property
+    def cv_parlist(self):
+        '''Construct the parameter list needed by the CV'''
+
+        if self.iscomplex:
+            par_name_list = (
+                'wdFlux', 'dFlux', 'sFlux', 'rsFlux', 'q', 'dphi',
+                'rdisc', 'ulimb', 'rwd', 'scale', 'az', 'fis', 'dexp', 'phi0',
+                'exp1', 'exp2', 'tilt', 'yaw'
+            )
+        else:
+            par_name_list = (
+                'wdFlux', 'dFlux', 'sFlux', 'rsFlux', 'q', 'dphi',
+                'rdisc', 'ulimb', 'rwd', 'scale', 'az', 'fis', 'dexp', 'phi0'
+            )
+
+        param_dict = self.ancestor_param_dict
+        parlist = [param_dict[key].currVal for key in par_name_list]
+
+        return parlist
+
+    @property
+    def iscomplex(self):
+        '''True if the eclipse uses the complex model. False otherwise'''
+        return len(self.node_par_names) == 13
+
+
+class Band(Model):
+    '''Subclass of Model, specific to observation bands. Contains the eclipse
+    objects taken in this band.
+
+    Inputs:
+    -------
+      label; str:
+        A label to apply to the node. Mostly used when searching trees.
+      parameter_objects; list(Param), or Param:
+        The parameter objects that correspond to this node. Single Param is
+        also accepted.
+      parent; Model, optional:
+        The parent of this node.
+      children; list(Model), or Model:
+        The children of this node. Single Model is also accepted
+    '''
+
+    # What kind of parameters are we storing here?
+    node_par_names = ('wdFlux', 'rsFlux')
+
+
+class LCModel(Model):
+    '''Top layer Model class. Contains Bands, which contain Eclipses.
+    Inputs:
+    -------
+      label; str:
+        A label to apply to the node. Mostly used when searching trees.
+      parameter_objects; list(Param), or Param:
+        The parameter objects that correspond to this node. Single Param is
+        also accepted.
+      parent; Model, optional:
+        The parent of this node.
+      children; list(Model), or Model:
+        The children of this node. Single Model is also accepted
+    '''
+
+    # Set the parameter names for this layer
+    node_par_names = ('q', 'dphi', 'rwd')
+
+    def chisq(self, verbose=False):
```

Comment:
If we do this with some care, this shouldn't be needed. The abstract ```Model``` class should have a method called ```chisq``` which just sums the value returned by all the children's chisq. Then you over-ride chisq on a leaf node to return a value, and bob is your uncle.

The same logic can be used for ln_like etc; and you only need to over-ride ln_like, chisq, evaluate on leaf nodes - all other nodes inherit the default from Model

Reply:
Done. Very clean